drivers/net/wireless/marvell/mwifiex/cfg80211.c

```diff
@@ -962,6 +962,138 @@ mwifiex_init_new_priv_params(struct mwifiex_private *priv,
 	return 0;
 }
 
+static bool
+is_vif_type_change_allowed (struct mwifiex_adapter *adapter,
+			    enum nl80211_iftype old_iftype,
+			    enum nl80211_iftype new_iftype)
+{
+	switch (old_iftype) {
+	case NL80211_IFTYPE_ADHOC:
+		switch (new_iftype) {
+		case NL80211_IFTYPE_STATION:
+			return true;
+		case NL80211_IFTYPE_P2P_CLIENT:
+		case NL80211_IFTYPE_P2P_GO:
+			return adapter->curr_iface_comb.p2p_intf ==
+			       adapter->iface_limit.p2p_intf;
```

Comment:
Are these checks correct? In the original code you have
```c
if (adapter->curr_iface_comb.p2p_intf ==
    adapter->iface_limit.p2p_intf) {
	mwifiex_dbg(adapter, ERROR,
		    "cannot create multiple P2P ifaces\n");
	return -1;
}
```
so shouldn't this be
```c
return adapter->curr_iface_comb.p2p_intf !=
       adapter->iface_limit.p2p_intf;
```
instead?

Reply:
Ouch, nice catch, indeed! That's what reviews are for, aren't they ;)

Reply:
Fixed that now